src/core/thread/mesh_forwarder.cpp

```diff
@@ -1014,6 +1033,18 @@ void MeshForwarder::HandleSentFrame(Mac::Frame &aFrame)
         {
             child->mFragmentOffset = 0;
             mSendMessage->ClearChildMask(mMle.GetChildIndex(*child));
+
+            if ((child->mMode & Mle::ModeTlv::kModeRxOnWhenIdle) == 0)
+            {
+                mMac.ClearSrcMatchAddressEntry(macDest);
+
```

Comment:
Whenever we have a message for a dst we invoke `AddSrcMatchAddressEntry` to notify the radio driver of the pending message for the dst, and we clear the address by calling `ClearSrcMatchAddressEntry` from the `HandleSentFrame` (when the frame is sent).

What if we get two messages queued for the same destination? Then we will clear the address when the first message is sent while we still have the second one in queue, can this cause a problem?